### DISCUSSION TITLE
How to consume messages in last N days?

### DISCUSSION THREAD

**Discussion by tigerinus** [confluentinc/confluent-kafka-python#1283] [Q&A]
I looked into the [Consumer.offsets_for_times](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#confluent_kafka.Consumer.offsets_for_times) function but I'm confused by that it accepts timestamps in the TopicPartition.offset field.

How is a offset equivalent to a timestamp?

Thanks.

**Comment by edenhill** [comment]
The offset field is reused for timestamp, so with offsets_for_times() you pass the timestamp (milliseconds since epoch) in the .offset field, and it will return the actual offset in the returned .offset field.
